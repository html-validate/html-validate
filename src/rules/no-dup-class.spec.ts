import HtmlValidate from '../htmlvalidate';

describe('rule no-dup-class', function() {
	let htmlvalidate: HtmlValidate;

	beforeAll(function() {
		htmlvalidate = new HtmlValidate({
			rules: { 'no-dup-class': 'error' },
		});
	});

	it('should not report when class is missing', function() {
		const report = htmlvalidate.validateString('<p></p>');
		expect(report).toBeValid();
	});

	it('should not report when class has no duplicates', function() {
		const report = htmlvalidate.validateString('<p class="foo bar"></p>');
		expect(report).toBeValid();
	});

	it('should report when when class has duplicates', function() {
		const report = htmlvalidate.validateString('<p class="foo bar foo"></p></p>');
		expect(report).toBeInvalid();
		expect(report).toHaveError('no-dup-class', 'Class "foo" duplicated');
	});

	it('smoketest', () => {
		const report = htmlvalidate.validateFile('test-files/rules/no-dup-class.html');
		expect(report.results).toMatchSnapshot();
	});

});
