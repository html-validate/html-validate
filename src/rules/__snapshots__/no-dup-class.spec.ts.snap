// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule no-dup-class should contain documentation 1`] = `
Object {
  "description": "Prevents unnecessary duplication of class names.",
  "url": "https://html-validate.org/rules/no-dup-class.html",
}
`;

exports[`rule no-dup-class smoketest 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "test-files/rules/no-dup-class.html",
    "messages": Array [
      Object {
        "column": 13,
        "context": undefined,
        "line": 5,
        "message": "Class \\"foo\\" duplicated",
        "offset": 66,
        "ruleId": "no-dup-class",
        "selector": "div:nth-child(3)",
        "severity": 2,
        "size": 11,
      },
    ],
    "source": "<div class=\\"foo\\"></div>

<div class=\\"foo bar\\"></div>

<div class=\\"foo bar foo\\"></div>
",
    "warningCount": 0,
  },
]
`;
